mailpoet/lib/Cron/Workers/Mixpanel.php

```diff
@@ -0,0 +1,64 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Cron\Workers;
+
+use MailPoet\Analytics\Analytics;
+use MailPoet\Entities\ScheduledTaskEntity;
+use MailPoet\WP\Functions;
+use Mixpanel as MixpanelLibrary;
+
+class Mixpanel extends SimpleWorker {
+
+  const PRODUCTION_PROJECT_ID = '8cce373b255e5a76fb22d57b85db0c92';
+
+  /** @var Analytics */
+  private $analytics;
+
+  const TASK_TYPE = 'mixpanel';
+
+  /** @var MixpanelLibrary */
+  private $mixpanel;
+
+  public function __construct(
+    Analytics $analytics,
+    Functions $wp
+  ) {
+    parent::__construct($wp);
+    $this->analytics = $analytics;
+    $this->mixpanel = MixpanelLibrary::getInstance(self::PRODUCTION_PROJECT_ID);
+    $this->mixpanel->register('Platform', 'Plugin');
+  }
+
+  public function processTaskStrategy(ScheduledTaskEntity $task, $timer) {
+    return $this->maybeReportAnalyticsToMixpanel();
+  }
+
+  public function maybeReportAnalyticsToMixpanel(): bool {
+    if (!$this->analytics->shouldSend()) {
+      return true;
+    }
+    return $this->reportAnalyticsToMixpanel();
+  }
+
+  public function reportAnalyticsToMixpanel(): bool {
+    $publicId = $this->analytics->getPublicId();
```

Comment:
If I see this correctly we rely on cookies for the public id. Maybe this comes later, but where do we `set` the public ID and have access to cookies (as this is a cron job here)?

Reply:
https://github.com/mailpoet/mailpoet/blob/8fbac3e2c25328a5b967e2f9c074e24caf0f0381/mailpoet/lib/Analytics/Analytics.php#L60-L71

Reply:
Thanks for flagging this, this is a valid concern!

I think it's unlikely we would end up in a situation where would need to rely on the cookie in the cron task. If a user is opted in to analytics and 3rd party libraries, we load `analytics.html` on every MailPoet page (https://github.com/mailpoet/mailpoet/blob/6e705fb316e8b2099036707ba37444928addfa70/mailpoet/views/layout.html#L48-L48) so we would end up with a value in the settings table.

The problem would be if the user isn't opted in to 3rd party libraries, in which case this never loads and we never get the public id.

I believe the only other time we explicitly set public id is if we get one back from the MSS:
https://github.com/mailpoet/mailpoet/blob/6e705fb316e8b2099036707ba37444928addfa70/mailpoet/lib/API/JSON/v1/Services.php#L131-L131
https://github.com/mailpoet/mailpoet/blob/6e705fb316e8b2099036707ba37444928addfa70/mailpoet/lib/API/JSON/v1/Services.php#L197-L197

We could end up with empty public IDs now, but I don't think we would ever try to use them if since we wouldn't load `analytics.html`.

It doesn't seem like the actual ID matters at all if we're using an md5, so my inclination is to simply set a random value instead of relying on the cookie. How does that sound to you? I'm going to check with the team just to make sure I'm not missing anything.

Reply:
943e1a64ef9c01cf251978bd21c51e9a341fd516 modifies this to use a random string instead of relying on the cookie. Under the hood it looks like this is basically what mixpanel does anyway: https://github.com/mixpanel/mixpanel-js/blob/8ac526e5cb8563d11e2206046ab986c6491ac6d7/src/mixpanel-core.js#L323C1-L333C6

Reply:
Yeah, in general I agree, it seems we can just set an ID ourselves.